### ISSUE TITLE
Allow for more streamlined vault provider integration

### ISSUE THREAD

**Issue by rgbrizzlehizzle** [hashicorp/terraform-provider-hcp#133]
### Description

Hey, cannot tell you how excited I am for this provider!  I have a small QoL feature request that I think people will like.  

Right now, the `hcp_vault_cluster.vault_public_endpoint_url` is mistmatched with what a standard vault provider would expect.  Users need to do something like this 

```hcl
provider "vault" {
  address = join("", ["https://", hcp_vault_cluster.bkbn.vault_public_endpoint_url, ":8200"])
  token = hcp_vault_cluster_admin_token.bkbn.token
}
```

in order to point a vault provider at a HCP vault cluster.

Ideally, this should be streamlined to something like

```hcl
 provider "vault" {
  address = hcp_vault_cluster.bkbn.vault_public_full_url // Not a fan of this name but you get the point 😃 
  token = hcp_vault_cluster_admin_token.bkbn.token
}
```

### New or Affected Resource(s)

* hcp_0.6.0

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by bcmdarroch** [comment]
Thanks for submitting this issue @rgbrizzlehizzle! I'm currently investigating whether we can fix this at the API level, so will keep you posted!

**Comment by rgbrizzlehizzle** [comment]
closing, resolved in hcp provider v `0.8.0`, thanks @bcmdarroch 🙏
